src/ir/module-splitting.cpp

```diff
@@ -369,17 +429,57 @@ void ModuleSplitter::setupTablePatching() {
     }
   });
 
+  if (replacedElems.size() == 0) {
+    // No placeholders to patch out of the table
+    return;
+  }
+
+  if (tableManager.activeBase.global) {
+    assert(primary.table.segments.size() == 1 &&
+           "Unexpected number of segments with non-const base");
+    assert(secondary.table.segments.size() == 0);
+    // Since addition is not currently allowed in initializer expressions, we
+    // need to start the new secondary segment where the primary segment starts.
+    // The secondary segment will contain the same primary functions as the
+    // primary module except in positions where it needs to overwrite a
+    // placeholder function. All primary functions in the table therefore need
+    // to be imported into the second module. TODO: use better strategies here,
+    // such as using ref.func in the start function or standardizing addition in
+    // initializer expressions.
```

Comment:
In theory we could only export functions in the gap slots, not all functions within the segment. But I'm not sure this is an important enough thing to optimize. (Does exporting / importing more functions cost us more?)

Reply:
This early exit is what the `&& replacement != replacedElems.end()` condition in the loop below is doing. You can see this tested in module-splitting.txt. Search for "(elem (global.get $base) $foo $bar $baz)" and notice that it does not include $quux in the secondary segment.

My impression is that exporting and importing isn't too expensive, but we still want to avoid it where we can for code size.